Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -423,9 +424,10 @@ mkDriverFee now merchantId driverId rideFare govtCharges platformFee cgst sgst t
         updatedAt = now,
         platformFee = platformFee_,
         totalEarnings = fromMaybe 0 rideFare,
-        feeType = case plan of
-          Nothing -> DF.RECURRING_INVOICE
-          Just plan_ -> if plan_.paymentMode == MANUAL then DF.RECURRING_INVOICE else DF.RECURRING_EXECUTION_INVOICE,
+        feeType = case (plan, isPlanMandatory) of
+          (Nothing, _) -> DF.RECURRING_INVOICE
```

Comment:
Plan can be nothing when isPlanMandatory is true here? If else can be used if not

Reply:
Handled the case just in case it occurs.